cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -521,56 +498,186 @@ def find_existing_views(self, client: CogniteClient) -> set:
+        """
+        if model := client.data_modeling.data_models.retrieve((self.space, self.external_id, self.version)):
+            cdf_data_model = model.latest_version()
+            return cdf_data_model.as_id()
+        return None
+
+    def to_cdf(
+        self,
+        client: CogniteClient,
+        components_to_create: set | None = None,
+        existing_component_handling: Literal["fail", "skip", "update"] = "fail",
+    ) -> None:
+        """Write the the data model to CDF.
+
+        Args:
+            client: Connected Cognite client.
+            components_to_create: Which components to create. Takes set
+            existing_component_handling: How to handle existing components. Takes Literal["fail", "skip", "update"]
+
+        !!! note "Component Creation Policy"
+            Here is more information about the different component creation policies:
+            - `all`: all components of the data model will be created, meaning space, containers, views and data model
+            - `data model`: only the data model will be created
+            - `view`: only the views will be created
+            - `container`: only the containers will be created
+
+
+        !!! note "Existing Component Handling Policy"
+            Here is more information about the different existing component handling policies:
+            - `fail`: if any component of the data model (DMS schema) already exists
+            - `skip`: skip DMS components that exist
+            - `update`: create DMS components that do not exist and update those that do exist
+
+            `update` policy is currently not implemented !
+
+        """
+
+        if components_to_create is None:
+            components_to_create = set(["all"])
+        existing_spaces = self.find_existing_spaces(client)
+        existing_containers = self.find_existing_containers(client)
+        existing_views = self.find_existing_views(client)
+        existing_data_model = self.find_existing_data_model(client)
+
+        if (
+            existing_spaces or existing_containers or existing_data_model or existing_views
+        ) and existing_component_handling == "fail":
+            raise exceptions.DataModelOrItsComponentsAlreadyExist(
+                existing_spaces, existing_data_model, existing_containers, existing_views
+            )
+
+        if "space" in components_to_create or "all" in components_to_create:
+            self.create_space(client, existing_spaces)
+        if "container" in components_to_create or "all" in components_to_create:
+            self.create_containers(client, existing_containers)
+        if "view" in components_to_create or "all" in components_to_create:
+            self.create_views(client, existing_views)
+        if "data model" in components_to_create or "all" in components_to_create:
+            self.create_data_model(client, existing_data_model)
+
+    def create_space(self, client: CogniteClient, existing_spaces: set | None = None):
```

Comment:
Lets rewrite this function to do it in one request. Not one request per space.

Reply:
done!